p_sets/3/find.rb

```diff
@@ -0,0 +1,56 @@
+module Finder
+  def self.linear_search(needle, hay)
+    max_index   = hay.length - 1
+    result      = false
+    max_index.times do |i|
+      result = true if hay[i] == needle
+    end
+    puts result.to_s
+  end
+
+  def self.binary_search(needle, hay)
+    mid_point = (hay.length - 1) / 2
+    max_point = hay.length - 1
+    left_side = 0..mid_point
+    right_side = (mid_point + 1)..max_point
+
+    if needle < hay[0] || needle > hay[max_point]
+      result = false
+    else
+      if needle < hay[mid_point]
+        binary_search(needle, hay[left_side])
+      elsif needle > hay[mid_point]
+        binary_search(needle, hay[right_side])
+      else
+        result = hay[mid_point]
```

Comment:
I think this prints the wrong thing. If I did `binary_search([10,11,12], 12)` it would print 12 rather than 2.